py2app/recipes/black.py

```diff
@@ -0,0 +1,25 @@
+from pkg_resources import get_distribution
+from pathlib import Path
```

Comment:
The upcoming release of py2app will be the last one that still somewhat supports Python 2.7, and python 2.7 does not support pathlib. Post-release I'll update the distribution metadata for py2app to require at least python 3.6.

That said, if you don't mind that this PR gets pulled after the upcoming release this code is fine (and nicer than code using the older os.path interfaces)

Reply:
I'd really like this PR to be pulled before the upcoming release, so I'll eliminate `pathlib`.